sn_networking/src/event.rs

```diff
@@ -967,9 +968,11 @@ impl SwarmDriver {
         if let Some((sender, result_map, quorum, expected_holders)) =
             self.pending_get_record.remove(query_id)
         {
-            if let (true, Ok(true)) = (
+            if let (true, Ok(true), true) = (
                 expected_holders.is_empty(),
                 RecordHeader::is_record_of_type_chunk(&peer_record.record),
+                // Ensure that we only exit early if quorum is indeed for only one match
+                matches!(quorum, GetQuorum::One),
```

Comment:
```suggestion
            if expected_holders.is_empty() &&
               RecordHeader::is_record_of_type_chunk(&peer_record.record).unwrap_or(false) &&
               // Ensure that we only exit early if quorum is indeed for only one match
               matches!(quorum, GetQuorum::One)
```

Reply:
```rust
 if let (true, Ok(true), true) = (
```
is a bit clunky no?

Reply:
I was just rolling with what we had here, but worth a refactor indeed 👍